### PR TITLE
added invert_match optional arg in core.contactmap.find

### PR DESCRIPTION
**Pull request by jjavier-bm**

### CODE REVIEW

File: conkit/core/contactmap.py [hunk 1]
```diff
@@ -655,7 +655,9 @@ def find(self, register, altloc=False, strict=False):
            Use the :attr:`~conkit.core.contact.Contact.res_altloc` positions [default: False]
         strict : bool
            Both residues of :obj:`~conkit.core.contact.Contact` in register [default: False]
-
+        invert_match : bool
```
Comment (fsimkovic) [inline conkit/core/contactmap.py]: I think `invert` is sufficient.

File: conkit/core/contactmap.py [hunk 2]
```diff
@@ -644,7 +644,7 @@ def set_scalar_score(self):
         for contact, sca_score in zip(self, sca_scores):
             contact.scalar_score = sca_score
 
-    def find(self, register, altloc=False, strict=False):
+    def find(self, register, altloc=False, strict=False, invert=False):
```
Comment (fsimkovic) [inline conkit/core/contactmap.py]: thinking again, let's call this `inverse`

**Comment by fsimkovic** [comment]
Thanks @jjavier-bm for the code change. I added a diff below showing the change that I think should be done. A couple of points
- test cases should be added to ensure functionality is correct
- instead of extending if statements, I added a little wrapper that should handle the inverse
- renamed invert to inverse
```
diff --git a/conkit/core/contactmap.py b/conkit/core/contactmap.py
index 7c1f4cc..a1691a5 100644
--- a/conkit/core/contactmap.py
+++ b/conkit/core/contactmap.py
@@ -644,7 +644,7 @@ class ContactMap(Entity):
         for contact, sca_score in zip(self, sca_scores):
             contact.scalar_score = sca_score
 
-    def find(self, register, altloc=False, strict=False):
+    def find(self, register, altloc=False, strict=False, inverse=False):
         """Find all contacts with one or both residues in ``register``
 
         Parameters
@@ -655,6 +655,8 @@ class ContactMap(Entity):
            Use the :attr:`~conkit.core.contact.Contact.res_altloc` positions [default: False]
         strict : bool
            Both residues of :obj:`~conkit.core.contact.Contact` in register [default: False]
+        invert_match : bool
+           Select non-matching residues [default: False]
 
         Returns
         -------
@@ -666,11 +668,17 @@ class ContactMap(Entity):
         if isinstance(register, int):
             register = [register]
         register = set(register)
-        comparison_operator = operator.and_ if strict else operator.or_
+
+        def keep(*args, **kwargs):
+            output = (operator.and_ if strict else operator.or_)(*args, **kwargs)
+            return output if not inverse else not output
+
         contact_map = self.deepcopy()
         for contactid in self.as_list(altloc=altloc):
-            if not comparison_operator(contactid[0] in register, contactid[1] in register):
-                contact_map.remove(tuple(contactid))
+            if keep(contactid[0] in register, contactid[1] in register):
+                continue
+            contact_map.remove(tuple(contactid))
+
         return contact_map
 
     def match(
diff --git a/conkit/core/tests/test_contactmap.py b/conkit/core/tests/test_contactmap.py
index 0529802..7486e13 100644
--- a/conkit/core/tests/test_contactmap.py
+++ b/conkit/core/tests/test_contactmap.py
@@ -385,6 +385,24 @@ class TestContactMap(unittest.TestCase):
         self.assertEqual([1, 2], [c.res1_seq for c in found])
         self.assertEqual([5, 5], [c.res2_seq for c in found])
 
+    def test_find_inverse(self):
+        contact_map1 = ContactMap("1")
+        for comb in [(1, 5, 1.0), (2, 6, 1.0), (1, 4, 1.0), (3, 6, 1.0), (2, 5, 1.0)]:
+            contact_map1.add(Contact(*comb))
+        found = contact_map1.find([1, 2, 5], strict=False, inverse=True)
+        self.assertEqual(1, len(found))
+        self.assertEqual([3], [c.res1_seq for c in found])
+        self.assertEqual([6], [c.res2_seq for c in found])
+
+    def test_find_strict_inverse(self):
+        contact_map1 = ContactMap("1")
+        for comb in [(1, 5, 1.0), (2, 6, 1.0), (1, 4, 1.0), (3, 6, 1.0), (2, 5, 1.0)]:
+            contact_map1.add(Contact(*comb))
+        found = contact_map1.find([1, 2, 5], strict=True, inverse=True)
+        self.assertEqual(3, len(found))
+        self.assertEqual([2, 1, 3], [c.res1_seq for c in found])
+        self.assertEqual([6, 4, 6], [c.res2_seq for c in found])
+
     def test_set_sequence_register_1(self):
         contact_map1 = ContactMap("1")
         for comb in [(1, 5, 1.0), (2, 6, 1.0), (1, 4, 1.0), (3, 6, 1.0), (2, 5, 1.0)]:
```

**Comment by jjavier-bm** [comment]
Thank you for your comments @fsimkovic.

- I have included the tests to the tests_contactmap.py file
- I have slightly modified the code you suggested for contactmap.find() to eliminate an excess of "not" operators that made following code a bit confusing.

Let me know what you think about it. If you don't like it I will revert it immediately.

**Comment by fsimkovic** [comment]
lgtm. thanks for the contribution 👍

**Pull request merged**